rfcs/0009-defaults-and-overrides.md

```diff
@@ -0,0 +1,947 @@
+## Status reporting and Policy discoverability
+
+### Possible statuses of an inherited policy
+
+An inherited policy can be at any of the following conditions ([RFC 0004](https://docs.kuadrant.io/architecture/rfcs/0004-policy-status/)):
+
+| Type     | Status | Reason                           | Message                                                                                                                                                                                                                 |
+|:--------:|:------:|----------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Accepted | True   | "Accepted"                       | "Policy has been accepted"                                                                                                                                                                                              |
+|          | False  | "Conflicted"                     | "Policy is conflicted by &lt;policy-ns/policy-name&gt;"                                                                                                                                                                 |
+|          | False  | "Invalid"                        | "Policy is invalid"                                                                                                                                                                                                     |
+|          | False  | "TargetNotFound"                 | "Policy target &lt;resource-name&gt; was not found"                                                                                                                                                                     |
+| Enforced | True   | "Enforced"                       | "Policy has been successfuly enforced"                                                                                                                                                                                  |
+|          | True   | "EnforcedWithAdditions"          | "Policy has been successfuly enforced. The following defaults have been added by &lt;policy-ns/policy-name&gt;: x, y"                                                                                                   |
+|          | True   | "PartiallyEnforced"              | "Policy has been successfuly enforced. The following rules have been overridden by &lt;policy-ns/policy-name&gt;: a, b"                                                                                                 |
+|          | True   | "PartiallyEnforcedWithAdditions" | "Policy has been successfuly enforced. The following rules have been overridden by &lt;policy-ns/policy-name&gt;: a, b; the following defaults have been added by &lt;policy-ns/policy-name&gt;: x, y"                  |
```

Comment:
I thought about adding these 2 extra reasons to cover the inheritance ("addition"), at the lower level, of policy rules defined at the upper levels.

An alternative, to not deviate from [RFC 0004](https://docs.kuadrant.io/architecture/rfcs/0004-policy-status), could be working with only "Enforced" and "PartiallyEnforced" and add the details of the additions to the message. I.e.:

| Type     | Status | Reason                           | Message                                                                                                                                                                                                  |
|:--------:|:------:|----------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Enforced | True   | "Enforced"                       | "Policy has been successfuly enforced[. The following defaults have been added by &lt;policy-ns/policy-name&gt;: x, y]"                                                                                  |
|          | True   | "PartiallyEnforced"              | "Policy has been successfuly enforced. The following rules have been overridden by &lt;policy-ns/policy-name&gt;: a, b[; the following defaults have been added by &lt;policy-ns/policy-name&gt;: x, y]" |
|          | False  | "Overridden"                     | "Policy has been overridden by &lt;policy-ns/policy-name&gt;"                                                                                                                                            |
|          | False  | "Unknown"                        | "Policy has encountered some issues"                                                                                                                                                                     |

cc @didierofrivia

Reply:
I'm also leaning towards using the current "Enforced" and "PartitallyEnforced" statuses with detailed messages to keep it a bit simpler